firefox_puppeteer/base.py

```diff
@@ -28,16 +28,44 @@ def get_marionette(self):
 class UIBaseLib(BaseLib):
     """A base class for all UI element wrapper classes inside a chrome window."""
 
-    def __init__(self, marionette_getter, window):
-        BaseLib.__init__(self, marionette_getter)
-
+    def __init__(self, marionette_getter, parent, element):
```

Comment:
Every element in the DOM has a parent, except for the top level root node. That's why the check for the `BaseWindow` instance. I never thought that we copy the DOM 1:1, and I don't think that this is our intention. We are building up a tree of UI elements, which are important for our testing. The DOM is not relevant here IMHO.